### PR TITLE
applications: nrf_desktop: Improve memory layout documentation

### PR DESCRIPTION
**Pull request by MarekPieta**
Change improves memory layout configuration documentation to simplify gettings started for new users.

Jira: NCSDK-24518

### CODE REVIEW

**Review by mkapala-nordic — APPROVED** [review]

File: applications/nrf_desktop/README.rst [hunk 1]
```diff
@@ -1555,52 +1555,61 @@ This code snippet will change the **SPI1** interrupt priority from default ``1``
...
+For example, the nRF52 Series uses flash non-volatile memory represented by the flash device node (``&flash0``).
+Make sure to also update the DTS chosen nodes representing the code partition (``zephyr,code-partition``) and flash (``zephyr,flash``) if needed.
 
-Since the nRF Desktop application uses the partition manager when the bootloader is present, the partition definition from the DTS files is valid only for configurations without the bootloader.
-
-.. note::
-    If you wish to change the default flash memory layout of the board without editing board-specific files, edit the DTS overlay file.
-    The nRF Desktop application automatically adds the overlay file if the :file:`dts.overlay` file is present in the project's board configuration directory.
-    See more in the `Board configuration`_ section.
+If you wish to change the default memory layout of the board without editing board-specific files, edit the DTS overlay file.
```
Comment (annwoj) [inline applications/nrf_desktop/README.rst:1593]:
```suggestion
If you wish to change the default memory layout of the board without editing the board-specific files, edit the DTS overlay file.
```

File: applications/nrf_desktop/README.rst [hunk 2]
```diff
@@ -1555,52 +1555,61 @@ This code snippet will change the **SPI1** interrupt priority from default ``1``
...
 
+The memory layout is defined through one of the following methods:
+
+* `Memory layout in DTS`_
+* `Memory layout in Partition Manager`_
+
+By default, a Zephyr-based application defines memory layout in the DTS.
+If enabled, the :ref:`partition_manager` defines a new memory layout that is used instead of the memory layout defined in the DTS.
+The :kconfig:option:`CONFIG_PARTITION_MANAGER_ENABLED` Kconfig option value can be used to check if the Partition Manager is enabled in the current build.
+The option is automatically selected in case :ref:`ug_multi_image` feature is used to build the application together with a bootloader or nRF53 network core image.
```
Comment (kapi-no) [inline applications/nrf_desktop/README.rst:1582]: What is the nRF53 network core image? nRF53 network core has many potential images that you can use. Let's make this sentence more generic and less target specific

```suggestion
The option is automatically selected as part of the :ref:`ug_multi_image` feature to build the application with more than one image.
```

Comment (MarekPieta) [reply applications/nrf_desktop/README.rst:1582]: Ok, we can update it.

File: applications/nrf_desktop/README.rst [hunk 3]
```diff
@@ -1555,52 +1555,61 @@ This code snippet will change the **SPI1** interrupt priority from default ``1``
...
    If it is incompatible, erase the settings area before using the new firmware.
 
+The memory layout is defined through one of the following methods:
+
+* `Memory layout in DTS`_
+* `Memory layout in Partition Manager`_
+
+By default, a Zephyr-based application defines memory layout in the DTS.
+If enabled, the :ref:`partition_manager` defines a new memory layout that is used instead of the memory layout defined in the DTS.
+The :kconfig:option:`CONFIG_PARTITION_MANAGER_ENABLED` Kconfig option value can be used to check if the Partition Manager is enabled in the current build.
```
[the remark this answers is not in the document]
Comment (MarekPieta) [reply applications/nrf_desktop/README.rst:1581]: Already applied as fix from @peknis

File: applications/nrf_desktop/README.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (MarekPieta) [reply applications/nrf_desktop/README.rst:1582]: Slightly rephrased following @kapi-no's suggestion

File: applications/nrf_desktop/README.rst [hunk 4]
```diff
@@ -1555,52 +1555,61 @@ This code snippet will change the **SPI1** interrupt priority from default ``1``
...
+Memory layout
+=============
 
+You must define the memory layout as a part of the application configuration for a given board.
 The set of required partitions differs depending on configuration:
 
 * There must be at least one partition where the code is stored.
 * There must be one partition for storing :ref:`zephyr:settings_api`.
-* The bootloader, if enabled, will add additional partitions to the set.
+* The bootloader, if enabled, adds more partitions to the set.
```
Comment (peknis) [inline applications/nrf_desktop/README.rst:1566]:
```suggestion
* If the bootloader is enabled, it adds more partitions to the set.
```

File: applications/nrf_desktop/README.rst [hunk 5]
```diff
@@ -1555,52 +1555,61 @@ This code snippet will change the **SPI1** interrupt priority from default ``1``
...
+The option is automatically selected as part of the :ref:`ug_multi_image` feature to build the application with more than one image.
+Enabling the :ref:`nrf_desktop_bluetooth_guide_fast_pair` also results in using the Partition Manager.
+To store the Fast Pair Provisioning data, the Fast Pair integration in the |NCS| uses partition defined by the Partition Manager.
+
 Memory layout in DTS
 --------------------
 
-When using the flash memory layout in the DTS files, define the ``partitions`` child node in the flash device node (``&flash0``).
+In case you rely on a non-volatile memory layout described in DTS files, define the ``partitions`` child node under the DTS node that represents the non-volatile memory.
+For example, the nRF52 Series use non-volatile flash memory represented by the flash device node (``&flash0``).
```
Comment (annwoj) [inline applications/nrf_desktop/README.rst:1590]:
```suggestion
For example, the nRF52 Series devices use non-volatile flash memory represented by the flash device node (``&flash0``).
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by annwoj — APPROVED** [review]

**Review by peknis — APPROVED** [review]

**Comment by MarekPieta** [comment]
Accidentally pushed a commit with additional changes after review - removed it in the subsequent push (commit was not intended to be part of the PR).

**Pull request merged**